Doctrine/Orm/OrderFilter.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Dunglas\ApiBundle\Doctrine\Orm;
+
+use Doctrine\ORM\QueryBuilder;
+use Dunglas\ApiBundle\Api\ResourceInterface;
+use Dunglas\ApiBundle\Doctrine\Orm\FilterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\VarDumper\VarDumper;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Théo FIDRY <theo.fidry@gmail.com>
+ */
+class OrderFilter implements FilterInterface
+{
+    /**
+     * @var array List of properties by witch the collection can or cannot be ordered.
+     */
+    private $sourceList = [];
+
+    /**
+     * @var bool If true $this#sourcelist is used as a whitelist. Otherwise is used as a blacklist.
+     */
+    private $type = true;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'order';
+    }
+
+    /**
+     * @param array   $properties List of property names on which the filter will be enabled/disabled.
+     * @param boolean $type       If true properties are passed as a whitelist and as a blacklist otherwise.
+     */
+    public function __construct(array $properties = [], $type = true)
+    {
+        $this->sourceList = $properties;
+        $this->type       = $type;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Order collection by properties. The order of the ordered properties is the same as the order specified in the
+     * query.
+     */
+    public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request)
+    {
+        $this->applyFilter($resource, $queryBuilder, $request->query->get('order'));
+    }
+
+    /**
+     * Create query to order the collection by the properties passed.
+     * For each property passed, if the resource does not have such property or if the order value is different from
+     * `asc` or `desc` (case insensitive), the property is ignored.
+     *
+     * @param ResourceInterface $resource
+     * @param QueryBuilder      $queryBuilder
+     * @param array             $values       Array of properties as key and order value as value.
+     *
+     * @return QueryBuilder
+     */
+    protected function applyFilter(ResourceInterface $resource, QueryBuilder $queryBuilder, array $values)
+    {
+        $metadata      = $queryBuilder->getEntityManager()->getClassMetadata($resource->getEntityClass());
+        $fieldNames    = array_flip($metadata->getFieldNames());
+        $filterList    = [];
+
+        // Get list of properties on which the collection can be ordered by.
+        // Is the list of the resource properties which are in the source list if is in whitelist mode or are not if in
+        // blacklist mode.
+        foreach ($fieldNames as $property => $value) {
+            if ($this->type === in_array($property, $this->sourceList)) {
+                $filterList[] = $property;
+            }
+        }
+
+        // Build query
+        foreach ($values as $property => $order) {
```

Comment:
The query building and the creation of the list can be done in the same loop IMO (performance boost).